### ISSUE TITLE
MSVC build utf-8 source file possible solution

### ISSUE THREAD

**Issue by hongnod** [copperspice/copperspice#124]
Without BOM, VC take the source as ASCII, build will fail, says the source has return character.

**Comment by agserm** [comment]
Can you provide some more information about your build environment, compiler version, and what error message you are receiving? Which file is reporting an issue during compilation?

**Comment by topillar** [comment]
The Problem resolved after adding two lines into the top CMakeList.txt .
```
add_compile_options("$<$<C_COMPILER_ID:MSVC>:/utf-8>")
add_compile_options("$<$<CXX_COMPILER_ID:MSVC>:/utf-8>")
```

**Comment by topillar** [comment]
Create a related pull request, pls verify it.

**Comment by janwilmans** [comment]
We reviewed you pull request, however this is not the way we want to handle this.
In CopperSpice, source files should not contain any non-ASCII characters, please provide the filenames of the files you had the problem with, so we can fix the files.

**Comment by topillar** [comment]
That means I need to build another time to get which files have  non-ASCII characters. I recall they are most come from third-party libs, such as Webkit.

**Comment by janwilmans** [comment]
@topillar, we have decided to use ASCII as the encoding for all source code in this repository, so utf-8 is not acceptable. I would very much appreciate if you could do another build to see which files caused a problem for you. Or alternatively provide us with a script of how you executed the build, so we can check it ourselves. I just rebuild using MSVC without any problems, so I'm very interested what could be different in your work-flow to cause these problems.

**Comment by topillar** [comment]
The reason is my windows system is Chinese version, so the MSVC treate the source encoding as GB2312, which is extended ASCII by using one or two bytes to encode Chinese Character.
I build the repos following this guide: https://www.copperspice.com/docs/cs_overview/build-overview.html
nothing special.
I will try another time to produce the error.
But I still suggest utf-8.
Because
1 CS use third-part libs， that kind of source will prefer utf-8
2 utf-8 is modern than ASCII, and it is compatible with ASCII
:)

**Comment by janwilmans** [comment]
Thank you for your feedback, we will take this into consideration. So are you using MSVC + ninja ?
And this setting is set to Chinese  ?
![image](https://user-images.githubusercontent.com/5933444/69962922-5bf99f00-150f-11ea-96d2-febe858cd4f9.png)

**Comment by topillar** [comment]
yes

**Comment by topillar** [comment]
I tried another time, reproduce the error another time.
the log is here [:](https://github.com/topillar/buildcslog)

**Comment by janwilmans** [comment]
Thanks!
```
..\src\3rdparty\webkit\Source\WebCore\platform\DefaultLocalizationStrategy.cpp(323): error C2001: 常量中有换行符
..\src\3rdparty\webkit\Source\WebCore\platform\DefaultLocalizationStrategy.cpp(323): fatal error C1057: 宏扩展中遇到意外的文件结束```

and

```
..\src\gui\kernel\qwidget.h(297): note: 查看对正在编译的函数 模板 实例化“const QString &cs_typeName<QFlags<Qt::InputMethodHint>>(void)”的引用
[1399/3787] Building CXX object src\gui\CMakeFiles\CsGui.dir\__\3rdparty\harfbuzz\src\hb-aat-map.cc.obj
D:\Codelabs\copperspice-cs-1.6.1\src\3rdparty\harfbuzz\src\hb.hh(1): warning C4819: 该文件包含不能在当前代码页(936)中表示的字符。请将该文件保存为 Unicode 格式以防止数据丢失
D:\Codelabs\copperspice-cs-1.6.1\src\3rdparty\harfbuzz\src\hb.hh(1): warning C4819: 该文件包含不能在当前代码页(936)中表示的字符。请将该文件保存为 Unicode 格式以防止数据丢失```

Are the most interesting kind of messages

**Comment by janwilmans** [comment]
The chinese text says: 

```
\src\3rdparty\harfbuzz\src\hb.hh(1): warning C4819: 
This file contains characters that cannot be represented in the current code page (936). Please save the file in Unicode format to prevent data loss
```

**Comment by topillar** [comment]
yes.

**Comment by bgeller** [comment]
We were aware of the text in the HB header file which is causing a problem for you in Chinese.  This is a third party library and will be updated to a newer version by the end of this year.

From your log I can see the other files are in the third party libraries for Webkit and FreeType. The best way to resolve this is by updating the Webkit files by hand. We already have a newer version of FreeType in testing. 

Thanks for reporting these issues.  Please note that adding a BOM is not supported on most compilers and would break all Unix platforms.

**Comment by topillar** [comment]
Thanks for your reply, I don't mean to add BOM any more,  only save  source file as UTF-8 WITHOUT BOM, in MSVC condition, just add two lines into CMakeList.txt file after `if(MSVC)`
```
add_compile_options("$<$<C_COMPILER_ID:MSVC>:/utf-8>")
add_compile_options("$<$<CXX_COMPILER_ID:MSVC>:/utf-8>")
```
in this way, you don't have to modify the third party libraries, and you don't need BOM, simply and works.
Best Regards.

**Comment by agserm** [comment]
This issue is resolved with commit 5795983. Please reopen if you have additional concerns.
